File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/chat/util/CheckUserLogin.java
package io.twogether.nbe_5_7_2_02team.chat.util;

import static io.twogether.nbe_5_7_2_02team.global.response.error.ErrorCode.CHAT_MEMBER_NOT_LOGIN;

import io.twogether.nbe_5_7_2_02team.global.exception.ErrorException;
import io.twogether.nbe_5_7_2_02team.member.dao.MemberRepository;
import io.twogether.nbe_5_7_2_02team.member.domain.Member;

import lombok.RequiredArgsConstructor;

import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CheckUserLogin {

    private final MemberRepository memberRepository;

    public Member checkUserLogin(UserDetails userDetails) {
        if (userDetails == null) {
            throw new ErrorException(CHAT_MEMBER_NOT_LOGIN);
        }

        return memberRepository.findById(Long.parseLong(userDetails.getUsername()));
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/chat/util/CheckUserLogin.kt
package io.twogether.nbe_5_7_2_02team.chat.util

import io.twogether.nbe_5_7_2_02team.global.exception.ErrorException
import io.twogether.nbe_5_7_2_02team.global.response.error.ErrorCode
import io.twogether.nbe_5_7_2_02team.member.dao.MemberRepository
import io.twogether.nbe_5_7_2_02team.member.domain.Member
import lombok.RequiredArgsConstructor
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Component
import java.util.function.Supplier

@Component
@RequiredArgsConstructor
class CheckUserLogin (
    private val memberRepository: MemberRepository
) {

    fun checkUserLogin(userDetails: UserDetails?): Member {
        if (userDetails == null) {
            throw ErrorException(ErrorCode.CHAT_MEMBER_NOT_LOGIN)
        }

        return memberRepository
            .findById(userDetails.username.toLong())
            .orElseThrow(Supplier { ErrorException(ErrorCode.NOT_FOUND_MEMBER) })
    }
}
